testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergSparkCompatibility.java

```diff
@@ -519,17 +540,11 @@ public void testPartitionedByNestedFiled()
                         "  TBLPROPERTIES ('format-version'=2)",
                 sparkTableName));
 
-        assertQueryFailure(() -> onTrino().executeQuery("INSERT INTO " + trinoTableName + " VALUES (2, ROW('b'))"))
-                .hasMessageContaining("Partitioning by nested field is unsupported: parent.nested");
-        assertQueryFailure(() -> onTrino().executeQuery("UPDATE " + trinoTableName + " SET id = 2"))
-                .hasMessageContaining("Partitioning by nested field is unsupported: parent.nested");
-        assertQueryFailure(() -> onTrino().executeQuery("DELETE FROM " + trinoTableName))
-                .hasMessageContaining("Partitioning by nested field is unsupported: parent.nested");
-        assertQueryFailure(() -> onTrino().executeQuery("MERGE INTO " + trinoTableName + " t USING " + trinoTableName + " s ON (t.id = s.id) WHEN MATCHED THEN UPDATE SET id = 2"))
-                .hasMessageContaining("Partitioning by nested field is unsupported: parent.nested");
-        assertQueryFailure(() -> onTrino().executeQuery("ALTER TABLE " + trinoTableName + " EXECUTE OPTIMIZE"))
-                .hasMessageContaining("Partitioning by nested field is unsupported: parent.nested");
-
+        onTrino().executeQuery("INSERT INTO " + trinoTableName + " VALUES (2, ROW('b'))");
+        onTrino().executeQuery("UPDATE " + trinoTableName + " SET id = 2");
```

Comment:
Does this statement have any effect?

Reply:
Separated test cases for nested partitioned field.